docs(k8s): add managed service definition (#1876)
scaleway/docs-content: containers/kubernetes/reference-content/managed-kubernetes-service-definition.mdx

dates:
  validation: 2023-06-29
  posted: 2023-06-29
categories:
  - kubernetes
---

Scaleway Kapsule and Kosmos are managed Kubernetes services. Scaleway's managed Kubernetes service abstracts away the complexities of managing and operating a Kubernetes cluster, enabling developers to focus on application development and deployment while ensuring a reliable and scalable environment for running containerized workloads.

### What is a managed Kubernetes service?

In the context of Scaleway, a managed Kubernetes service refers to the Kapsule and Kosmos products. Scaleway handles the management and maintenance of the Kubernetes control plane, along with all the crucial core components required for the proper operation of the Kubernetes cluster.

### What components of Kubernetes are managed by Scaleway?

Scaleway manages the Kubernetes control plane (either Kapsule or Kosmos), which consists of various components responsible for managing the cluster, scheduling applications, and maintaining cluster state.

This includes components such as the control plane itself: etcd, API server, scheduler, cloud controller, and controller manager.

> Scaleway is also responsible for the operational upkeep of the operating system and system applications on the nodes. This includes essential components like CoreDNS, Kubeproxy, Container Networking Interface (CNI), and Container Storage Interface (CSI), which are vital for the optimal functioning of the Kubernetes cluster and its associated resources.

Author: ```suggestion
Scaleway is also responsible for the operational upkeep of the operating system and system applications on the nodes. This includes essential components like CoreDNS, Kubeproxy, Container Networking Interface (CNI), and Container Storage Interface (CSI), which are vital for the optimal functioning of the Kubernetes cluster and its associated resources. To ensure your nodes are always using the latest version of the underlying OS image it is recommended to limit their lifespan to a period of 30 days and to replace them regularly. 
```

---
meta:
  title: Managed Kubernetes service definition
  description: This page explains the definiton of the managed Kubernetes service at Scaleway
content:
  h1: Managed Kubernetes service definition
  paragraph: This page explains the definiton of the managed Kubernetes service at Scaleway
tags: kubernetes kubernetes-kapsule kapsule kosmos kubernetes-kosmos managed service definition
dates:
  validation: 2023-06-29
  posted: 2023-06-29
categories:
  - kubernetes
---

Scaleway Kapsule and Kosmos are managed Kubernetes services. Scaleway's managed Kubernetes service abstracts away the complexities of managing and operating a Kubernetes cluster, enabling developers to focus on application development and deployment while ensuring a reliable and scalable environment for running containerized workloads.

### What is a managed Kubernetes service?

> In the context of Scaleway, a managed Kubernetes service refers to the Kapsule and Kosmos products, with Scaleway handling the management and maintenance of the Kubernetes control plane, along with all the crucial core components required for the proper operation of the Kubernetes cluster.

Author: ```suggestion
In the context of Scaleway, a managed Kubernetes service refers to the Kapsule and Kosmos products. Scaleway handles the management and maintenance of the Kubernetes control plane, along with all the crucial core components required for the proper operation of the Kubernetes cluster.
```

### What components of Kubernetes are managed by Scaleway?

Scaleway manages the Kubernetes control plane (either Kapsule or Kosmos), which consists of various components responsible for managing the cluster, scheduling applications, and maintaining cluster state.

This includes components such as the control plane itself: etcd, API server, scheduler, cloud controller, and controller manager.

> Scaleway also is responsible for the operational upkeep of the operating system and system applications on the nodes. This includes essential components like CoreDNS, Kubeproxy, Container Networking Interface (CNI), and Container Storage Interface (CSI), which are vital for the optimal functioning of the Kubernetes cluster and its associated resources.

Reviewer: ```suggestion
Scaleway is also responsible for the operational upkeep of the operating system and system applications on the nodes. This includes essential components like CoreDNS, Kubeproxy, Container Networking Interface (CNI), and Container Storage Interface (CSI), which are vital for the optimal functioning of the Kubernetes cluster and its associated resources.
```

### What responsibilities do I have as a user of a managed Kubernetes service?

As a user, your responsibility lies primarily in deploying and managing the applications or services that you intend to run on the Kubernetes cluster.

You are responsible for any deployments made to the cluster and the associated configurations.

> Furthermore, it is expected that you use the appropriate interfaces for managing resources, such as `kubectl` or the Scaleway Kubernetes console. However, please note that if you make any modification to the pre-installed components, such as Kubelet, at the node or system level as mentioned earlier, the responsibility for managing those components will be transferred to you.

Reviewer 2: kubectl and the scaleway console/CLI are complementary not substitutable therefore I would use `and` instead of `or`

Author: ```suggestion
Furthermore, it is expected that you use the appropriate interfaces for managing resources, such as `kubectl` and the Scaleway Kubernetes console. However, please note that if you make any modification to the pre-installed components, such as Kubelet, at the node or system level as mentioned earlier, the responsibility for managing those components will be transferred to you.
```

> Furthermore, it is expected that you utilize the appropriate interfaces for managing resources, such as `kubectl` or the Scaleway Kubernetes console. However, please note that if you make any modifications to the pre-installed components, such as Kubelet, at the node or system level as mentioned earlier, the responsibility for managing those components will be transferred to you.

Reviewer: ```suggestion
Furthermore, it is expected that you use the appropriate interfaces for managing resources, such as `kubectl` or the Scaleway Kubernetes console. However, please note that if you make any modification to the pre-installed components, such as Kubelet, at the node or system level as mentioned earlier, the responsibility for managing those components will be transferred to you.
```

> ### Who is responsible for Instances, bare metal machines, persistent volumes, and load balancers attached to the cluster?

Reviewer: Aren't some of them product names that should be capitalized?

Author: these are not product names but Kubernetes services

> ### Who is responsible for Instances, bare metal machines, persistent volumes, and load balancers attached to the cluster?

Reviewer 3: ```suggestion
### Who is responsible for Instances, Bare Metal machines, persistent volumes, and Load Balancers attached to the cluster?
```

When it comes to resources managed exclusively by users through the relevant interfaces like kubectl or the Scaleway Kubernetes console, Scaleway takes on the responsibility.

Kubernetes takes charge of provisioning and orchestrating nodes, persistent volumes, and load balancers that are associated with the cluster.

Scaleway ensures the proper configuration of the cluster to handle these resources and seamlessly integrates them into the Kubernetes environment.

### What happens in the event of a cluster failure?

> If a cluster failure occurs, Scaleway possesses the ability to reset or "restore" the cluster to its default state. However, it is crucial to understand that resetting the cluster does not guarantee immediate perfect functionality.

Reviewer: ```suggestion
If a cluster failure occurs, Scaleway has the ability to reset or "restore" the cluster to its default state. However, it is crucial to understand that resetting the cluster does not guarantee immediate perfect functionality.
```

Certain factors like misconfigurations, applications deployed by the client, or breaking changes in Kubernetes might have contributed to the failure.

In such cases, Scaleway is committed to offering evidence and recommendations through observability alerts or audit logs. These resources can assist the client in investigating and resolving the failure, providing valuable insights into the root cause.

> 

Reviewer: Are these linebreaks expected?

> 
> 

Author: ```suggestion
```

---
meta:

>   title: Managed Kubernetes service definition

Reviewer 4: ```suggestion
  title: Managed Kubernetes service definition
```

Author: this seems to be the same?

Reviewer 4: One was in the title and the other in the description.

Reviewer 4: Nevermind. I was reading the previous text, which you had already fixed. I'm sorry.

>   description: This page explains the definiton of the managed Kubernetes service at Scaleway

Reviewer 4: ```suggestion
  description: This page explains the definition of the managed Kubernetes service at Scaleway
```
